docs/documentation/server_development/topics/providers.adoc

```diff
@@ -216,10 +216,13 @@ need to be retrieved with the usage of `KeycloakSessionFactory`. It is not recom
 
 === Registering provider implementations
 
-Providers are registered with the server by simply copying them to the `providers` directory.
+Providers are registered with the server by simply copying the JAR file to the `providers` directory.
 
 If your provider needs additional dependencies not already provided by Keycloak copy these to the `providers` directory.
 
+NOTE: Do not include resources or classes in your provider jars that conflict with built-in resources or classes. 
```

Comment:
Should we perhaps be more specific here to where the users need to look to and what's considered a built-in resource?

Reply:
application.properties is called out specifically in the next line. Unfortunately there's not a great way for users to look for overlap on their own. The obvious ones are in our quarkus runtime resources, but there's plenty more if you took a full listing of all the non-class files in all of the built-in jars.

Reply:
Perhaps we could point the users to the `lib/lib/main` dir?

Reply:
I'd to both. Any hint might be a big help to a desperate user. :)

Reply:
Ok, expanded on this with more details.